api/v1alpha1/kustomization_types.go

```diff
@@ -82,10 +83,9 @@ type KustomizationSpec struct {
 
 // WorkloadReference defines a reference to a Deployment, DaemonSet or StatefulSet.
 type WorkloadReference struct {
```

Comment:
Should it still be called WorkloadReference as there no longer is a limit on the type of resource to use for the health checks.

Reply:
See my comment on `CrossNamespaceObjectReference`